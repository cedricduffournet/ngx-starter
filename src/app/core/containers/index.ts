export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
