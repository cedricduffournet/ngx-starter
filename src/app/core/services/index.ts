export * from './http-progress.interceptor';
export * from './http.service';
