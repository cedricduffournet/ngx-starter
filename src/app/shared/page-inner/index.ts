export * from './page-inner.module';
