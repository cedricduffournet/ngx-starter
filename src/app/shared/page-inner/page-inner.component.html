<div class="px-3 py-3 {{ elemClass }}">
  <ng-content></ng-content>
</div>
