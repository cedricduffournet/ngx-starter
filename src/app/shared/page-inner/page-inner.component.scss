
:host{
    margin-bottom: 60px;
    display: block;
}