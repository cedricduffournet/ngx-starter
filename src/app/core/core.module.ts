import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { environment } from '@env/environment';
import {
  HttpClientModule,
  HttpClient,
  HTTP_INTERCEPTORS
} from '@angular/common/http';

import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ModalModule } from 'ngx-bootstrap/modal';
import { ToasterModule } from 'angular2-toaster';

import {
  ToasterComponent,
  HttpProgressBarComponent
} from '@app/core/containers';
import { HttpProgressInterceptor } from '@app/core/services/http-progress.interceptor';

const MODULES: any[] = [
  HttpClientModule,
  TranslateModule.forRoot({
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }),
  ModalModule.forRoot(),
  ToasterModule.forRoot(),
  CommonModule
];

export const COMPONENTS = [ToasterComponent, HttpProgressBarComponent];

@NgModule({
  imports: MODULES,
  declarations: COMPONENTS,
  exports: COMPONENTS,
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpProgressInterceptor,
      multi: true
    }
  ]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import only once in main AppModule.'
      );
    }
  }
}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${environment.prefixAsset}/assets/i18n/`,
    '.json'
  );
}
