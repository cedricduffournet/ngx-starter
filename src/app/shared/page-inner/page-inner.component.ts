import { Component, ChangeDetectionStrategy, Input } from '@angular/core';

@Component({
  selector: 'app-page-inner',
  templateUrl: './page-inner.component.html',
  styleUrls: ['./page-inner.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class PageInnerComponent {
  @Input() elemClass = '';
}
