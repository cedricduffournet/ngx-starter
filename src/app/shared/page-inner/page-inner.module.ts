import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { PageInnerComponent } from '@app/shared/page-inner/page-inner.component';

@NgModule({
  imports: [CommonModule],
  exports: [PageInnerComponent],
  declarations: [PageInnerComponent]
})
export class PageInnerModule {}
