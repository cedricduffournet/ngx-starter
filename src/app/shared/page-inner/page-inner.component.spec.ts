import { TestBed, ComponentFixture, async } from '@angular/core/testing';

import { PageInnerComponent } from '@app/shared/page-inner/page-inner.component';

describe('PageInnerComponent', () => {
  let fixture: ComponentFixture<PageInnerComponent>;
  let component: PageInnerComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [PageInnerComponent]
    });
    fixture = TestBed.createComponent(PageInnerComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should be created with css class', () => {
    component.elemClass = 'test-class';

    fixture.detectChanges();

    expect(fixture.nativeElement.querySelector('.test-class')).toBeTruthy();
  });
});
