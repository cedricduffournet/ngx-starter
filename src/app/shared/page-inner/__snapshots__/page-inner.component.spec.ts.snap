// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageInnerComponent should be created 1`] = `
<app-page-inner
  elemClass=""
>
  <div
    class="px-3 py-3 "
  />
</app-page-inner>
`;
